Content/Archaeology/Artifact.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Core;
+using StarlightRiver.Helpers;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.GameInput;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+using System;
+
+namespace StarlightRiver.Content.Archaeology
+{
+    public abstract class Artifact : ModTileEntity
+	{
+        public bool displayedOnMap = false; //Whether or not the artifact is displayed on the map
+
+        public virtual bool CanBeRevealed() => true; //Whether or not the artifact can be revealed by the archaeologist's map. Set to false if the artifact has a special reveal condition
+
+        public virtual string TexturePath => AssetDirectory.Archaeology + Name;
+
+        public virtual string MapTexturePath => AssetDirectory.Archaeology + "DigMarker"; //Texture path the artifact uses on the map when revealed
+
+        public virtual Vector2 Size { get; } //Size of the artifact. In world coordinates, not tile coordinates
+
+        public virtual int SparkleDust { get; } //The dust the artifact creates.
+
+        public virtual int SparkleRate { get; } //The rate at which sparkles spawn. Increase for lower spawnrate.
+
+        public virtual Color BeamColor { get; } //The color of the glowy effect when the artifact is excavated
+
+        public virtual int ItemType { get; } //The item the artifact drops
+
+        public virtual float SpawnChance { get; } //Pretty self explanatory. Higher = higher spawnrate
+
+        public Vector2 WorldPosition => Position.ToVector2() * 16;
+
+        public virtual bool CanGenerate(int i, int j) => true; //Override if you want to check at these specific coordinates whether the artifact can generate
```

Comment:
lets use summary comments here so documentation appears when hovering over this in VS

Reply:
Done